### PR TITLE
fixes #644 add cancellable content serializer

### LINKED ISSUE #644: Pass CancellationToken to IContentSerializer
I'm implementing IContentSerializer interface and in my implementation, I've no access to cancellation token. Could you please provide it too? Thanks in advance.

### PR DESCRIPTION
**Pull request by Styxxy**
fixes #644 add cancellable content serializer

I didn't add unit tests for now, as I don't know yet exactly how to test it. Any pointers are welcome  :).

I'll provide a separate PR for docs update (if applicable), so it can be merged after new version is being published.

### CODE REVIEW

File: Refit/RequestBuilderImplementation.cs [hunk 1]
```diff
@@ -215,7 +215,17 @@ async Task AddMultipartItemAsync(MultipartFormDataContent multiPartContent, stri
             Exception e = null;
             try
             {
-                multiPartContent.Add(await settings.ContentSerializer.SerializeAsync(itemValue).ConfigureAwait(false), parameterName);
+                HttpContent cont;
+                if (settings.ContentSerializer is IContentSerializerWithCancellation serializerWithCancellation)
+                {
+                    cont = await serializerWithCancellation.SerializeAsync(itemValue).ConfigureAwait(false);
```
Comment (onovotny) [inline Refit/RequestBuilderImplementation.cs]: This call is missing a cancellation token parameter

**Comment by onovotny** [comment]
For testing it, I'd suggest creating a mock content type that uses some kind of reset event (like `SepaphoreSlim(1,1)` in the serialize/deserialize and then use the passed-in cancellation token in a WaitOne call.

File: Refit/RequestBuilderImplementation.cs (hunk 1 above)
Comment (Styxxy) [reply Refit/RequestBuilderImplementation.cs]: Good catch, fixed it :)

**Comment by clairernovotny** [comment]
closing due to age. cancellation is passed into the deserialize method and so far none of the methods to create an HttpContent have required cancellation.

**Pull request closed without merging**